client/components/tinymce/plugins/wpcom-view/contact-form-view.jsx

```diff
@@ -32,10 +32,78 @@ export default React.createClass( {
 			editor.execCommand( 'wpcomContactForm', content );
 		}
 	},
+
+	renderContent() {
+		const required = required => required ? <em>&nbsp;({ this.translate( 'required' ) })</em> : null;
+		const textField = ( field, index ) => {
+			return (
+				<fieldset key={ 'contact-form-field-' + index } className="wpview-type-contact-form-field">
+					<legend>{ field.label }{ required( field.required ) }</legend>
+					<input type="text" />
+				</fieldset>
+			);
+		};
+		const textarea = ( field, index ) => {
+			return (
+				<fieldset key={ 'contact-form-field-' + index } className="wpview-type-contact-form-field">
+					<legend>{ field.label }{ required( field.required ) }</legend>
+					<textarea></textarea>
+				</fieldset>
+			);
+		};
+		const checkbox = ( field, index ) => {
+			return (
+				<fieldset key={ 'contact-form-field-' + index } className="wpview-type-contact-form-field">
+					<label><input type="checkbox" />{ field.label }{ required( field.required ) }</label>
+				</fieldset>
+			);
+		};
+		const dropdown = ( field, index ) => {
+			return (
+				<fieldset key={ 'contact-form-field-' + index } className="wpview-type-contact-form-field">
+					<legend>{ field.label }{ required( field.required ) }</legend>
+					<select>
+						{ field.options.map( ( option, index ) => (
+							<option key={ 'contact-form-select-option-' + index }>{ option }</option>
+						) ) }
+					</select>
+				</fieldset>
+			);
+		}
+		const radio = ( field, index ) => {
+			return (
+				<fieldset key={ 'contact-form-field-' + index } className="wpview-type-contact-form-field">
+					<legend>{ field.label }{ required( field.required ) }</legend>
+						{ field.options.map( ( option, index ) => (
+							<label key={ 'contact-form-radio-' + index }><input type="radio" /><span>{ option }</span></label>
+						) ) }
+				</fieldset>
+			);
+		}
+
+		const renderField = {
+			name: textField,
+			email: textField,
+			url: textField,
+			text: textField,
+			textarea,
+			checkbox,
+			dropdown
+		};
+
+		const { fields } = deserialize( this.props.content );
```

Comment:
If we coerce this into an array, we won't need to check to make sure it's an array in the next line:

```js
return [].concat( fields ).map( … )
```